### PR TITLE
AP_Math: rename macros to avoid conflicts

### PR DESCRIPTION
**Pull request by tridge**
these macros are also in NuttX clock.h. We need different names to avoid conflicts

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
Ugh... What about we remove/undef the define ? Let's not make our define look weird because of a corner case (since it wasn't a problem until now). 

The conditional undef can be on the autogenerated conf.h so it's guaranteed to be included by all compilation unjt

**Comment by tridge** [comment]
> What about we remove/undef the define ?

because in the case I hit, the NuttX define is 2nd. So we'd need to mod NuttX to do this. I know we can do that, but then we're building NuttX with slightly different defines for basis constants than it normally uses (we use ULL suffix, NuttX doesn't)

**Comment by tridge** [comment]
also, what's weird about MILLISEC that is weirder than MSEC ?
:-)

**Comment by lucasdemarchi** [comment]
> because in the case I hit, the NuttX define is 2nd. So we'd need to mod NuttX to do this.

no, we don't. See my proposal above about using the config.h header.

if we are renaming, then we can maintain consistency by prefixing with AP_

**Comment by lucasdemarchi** [comment]
> also, what's weird about MILLISEC that is weirder than MSEC ?
> :-)

this is derived from posix structs in which it uses "sec" and just one letter for abbreviation: n, m, u... and it's used on several other projects (including nuttx as you noticed ;-)).

**Comment by lucasdemarchi** [comment]
> because in the case I hit, the NuttX define is 2nd.

and realistically speaking you would include clock.h from NuttX in 1 or 2 place(s) only.... No need to redefine it everywhere else, unless we hit more of this which is unlikely.

**Comment by tridge** [comment]
> if we are renaming, then we can maintain consistency by prefixing with AP_

I'd prefer that. Conditionally undef relies on include order, which can be fragile, and could end up including NuttX macros or inline functions that rely on the exact version of the define
Using AP_ gets us away from those possibilities. I'll redo it that way

**Comment by dgrat** [comment]
These macros are just used a few times and maybe it would be better to define them in the *.cpp file instead in the library. Then all potential and future conflicts would be resolved. Alternatively, the proper way would to implement them as global variables in a namespace.

File: libraries/AP_Math/definitions.h [hunk 1]
```diff
@@ -64,10 +64,10 @@ static const double WGS84_B = (WGS84_A * (1 - WGS84_F));
...
-#define NSEC_PER_USEC   1000ULL
-#define USEC_PER_SEC    1000000ULL
-#define USEC_PER_MSEC   1000ULL
-#define MSEC_PER_SEC    1000ULL
-#define SEC_PER_WEEK    (7ULL * 86400ULL)
-#define MSEC_PER_WEEK   (SEC_PER_WEEK * MSEC_PER_SEC)
+#define NSEC_PER_SECOND   1000000000ULL
+#define NSEC_PER_USECOND  1000ULL
+#define USEC_PER_SECOND   1000000ULL
+#define USEC_PER_MILLISEC 1000ULL
```
Comment (EShamaev) [inline libraries/AP_Math/definitions.h]: USEC_PER_MILLISEC**OND** so it is at least consistent?

**Comment by tridge** [comment]
i've pushed a change to use AP_ prefix. I think it is the simplest solution. I don't want to turn this minor fix into a big deal.

**Pull request merged**